Sources/NIO/SelectorEpoll.swift

```diff
@@ -93,6 +93,35 @@ extension SelectorEventSet {
     }
 }
 
+@usableFromInline struct EPollUserData {
+    // We're backing this by a tuple to get standard C layout
+    @usableFromInline var registrationID: SelectorRegistrationID
+    @usableFromInline var fileDescriptor: CInt
+
+
+    @inlinable init(registrationID: SelectorRegistrationID, fileDescriptor: CInt) {
```

Comment:
Bit of a mild nit here, but all call sites of this appear to have `fileDescriptor` as an `Int` and so have to manually force it down to `CInt`. Should this init instead take `Int` and do the shrink here?

Reply:
Actually, I thought about file descriptors representation in general. Shouldn’t it be a type alias used consistently (which i believe should be CInt to allow for negative/invalid fd). It seems like an overkill to use Int, no?

Reply:
I don’t think the memory usage of `Int` in these spaces is an issue: either way it consumes a register, and in these cases it’s largely being passed around as an argument. However, I do agree that it would be nice to apply a type to it. In this case that might be a sensible follow-up proposal.

Reply:
Ok. I was not concerned about the memory usage/register usage, but rather to remove the need to shrink it when doing systemcalls. Basically all backends except a CInt, so it would remove the need to convert it ever.

Reply:
@Lukasa how strong are you on this one. I agree with @hassila that all these backends can only handle `CInt`. So if anything, we should refactor the existing code to do `CInt` instead. I can do this in this PR or another one (which I think should be preferred). WDYT?

Reply:
Another one is fine.

Reply:
awesome, thanks --> https://github.com/apple/swift-nio/pull/1815